config/ImathConfig.h.in

```diff
@@ -61,4 +61,18 @@
   #define IMATH_HOSTDEVICE
 #endif
 
+
+//
+// Simple way to mark things as deprecated
+//
+#if defined(__cplusplus) && (__cplusplus >= 201402L)
+# define IMATH_DEPRECATED(msg) [[deprecated(msg)]]
+#elif defined(_MSC_VER)
```

Comment:
It's so nice to see that _MSC_VER doesn't have to be detected early to bypass the historically broken __cplusplus check!!

Reply:
Oh, well when you say it like that, I wonder if I've done it wrong. There are some compilers that don't like

    #if defined(FOO) && ...expression involving FOO...

and I never quite remember when that's allowed. It's probably MSVC with the issue, right?

Perhaps I'll reorder this to be safe.

Reply:
It might depend on exactly which MSVC version it is. Hmmm. I'll fix to be sure it's 100% safe.

Reply:
Updated to be safer. Also caught one more file where things should have been changed to std::. Grep shows that's the last of it.